tests/e2e/kepler_test.go

```diff
@@ -149,3 +177,44 @@ func TestTaint_WithToleration(t *testing.T) {
 	f.AssertNoResourceExists(exporter.DaemonSetName, components.Namespace, &ds)
 
 }
+func TestBadTaint_WithToleration(t *testing.T) {
+
+	f := test.NewFramework(t)
+	// Ensure Kepler is not deployed (by any chance)
+	f.AssertNoResourceExists("kepler", "", &v1alpha1.Kepler{}, test.Timeout(10*time.Second))
+
+	// choose one node
+	nodes := f.GetSchedulableNodes()
+	node := nodes[0]
+	e2eTestTaint := corev1.Taint{
+		Key:    "key1",
+		Value:  "value1",
+		Effect: corev1.TaintEffectNoSchedule,
+	}
+	badTestTaint := corev1.Taint{
+		Key:    "key2",
+		Value:  "value2",
+		Effect: corev1.TaintEffectNoSchedule,
+	}
+
+	err := f.TaintNode(node.Name, e2eTestTaint.ToString())
+	assert.NoError(t, err, "failed to taint node %s", node)
+
+	f.CreateKepler("kepler", func(k *v1alpha1.Kepler) {
+		k.Spec.Exporter.Deployment.Tolerations = f.TolerateTaints(append(node.Spec.Taints, badTestTaint))
```

Comment:
May be a good idea to move this function out as `WithTolerations` so that the code reads 
```
f.CreateKepler("kepler", f.WithTolerations(...))
```